wp-content/themes/core/components/blocks/icon_grid/Icon_Grid_Controller.php

```diff
@@ -0,0 +1,251 @@
+use Tribe\Project\Templates\Components\text\Text_Controller;
+use Tribe\Project\Templates\Components\image\Image_Controller;
+use Tribe\Project\Theme\Config\Image_Sizes;
+
+class Icon_Grid_Controller extends Abstract_Controller {
+
+	public const CLASSES           = 'classes';
+	public const ATTRS             = 'attrs';
+	public const CONTAINER_CLASSES = 'container_classes';
+	public const CONTENT_CLASSES   = 'content_classes';
+	public const TITLE             = 'title';
+	public const LEADIN            = 'leadin';
+	public const DESCRIPTION       = 'description';
+	public const CTA               = 'cta';
+	public const ICONS             = 'icons';
+
+	private array  $classes;
+	private array  $attrs;
+	private array  $container_classes;
+	private array  $content_classes;
+	private string $title;
+	private string $leadin;
+	private string $description;
+	private array  $cta;
+	private array  $icons;
+
+	public function __construct( array $args = [] ) {
+		$args = $this->parse_args( $args );
+
+		$this->classes           = (array) $args[ self::CLASSES ];
+		$this->attrs             = (array) $args[ self::ATTRS ];
+		$this->container_classes = (array) $args[ self::CONTAINER_CLASSES ];
+		$this->content_classes   = (array) $args[ self::CONTENT_CLASSES ];
+		$this->title             = (string) $args[ self::TITLE ];
+		$this->leadin            = (string) $args[ self::LEADIN ];
+		$this->description       = (string) $args[ self::DESCRIPTION ];
+		$this->cta               = (array) $args[ self::CTA ];
+		$this->icons             = (array) $args[ self::ICONS ];
+	}
+
+	protected function defaults(): array {
+		return [
+			self::CLASSES           => [],
+			self::ATTRS             => [],
+			self::CONTAINER_CLASSES => [],
+			self::CONTENT_CLASSES   => [],
+			self::TITLE             => '',
+			self::LEADIN            => '',
+			self::DESCRIPTION       => '',
+			self::CTA               => [],
+			self::ICONS             => [],
+		];
+	}
+
+	protected function required(): array {
+		return [
+			self::CLASSES           => [ 'c-block', 'b-icon-grid' ],
+			self::CONTAINER_CLASSES => [ 'l-container' ],
+		];
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_classes(): string {
+		return Markup_Utils::class_attribute( $this->classes );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_attrs(): string {
+		return Markup_Utils::concat_attrs( $this->attrs );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_container_classes(): string {
+		return Markup_Utils::class_attribute( $this->container_classes );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_content_classes(): string {
+		$this->content_classes[] = 'g-3-up';
+		$this->content_classes[] = 'g-centered';
+
+		return Markup_Utils::class_attribute( $this->content_classes );
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_header_args(): array {
+		if ( empty( $this->title ) && empty( $this->description ) ) {
```

Comment:
Both title and description have to be empty for this not to proceed, that's the intention right?

Reply:
Correct. That's how we have all the other blocks set up currently.